src/w_wad.cc

```diff
@@ -162,7 +162,7 @@ int64_t Lump_c::getName8() const noexcept
 		char cbuf[8];
 		int64_t cint;
 	} buffer;
-	strncpy(buffer.cbuf, Name().c_str(), 8);
+	memcpy(buffer.cbuf, Name().c_str(), std::min(Name().length(), sizeof buffer));
```

Comment:
Looking deeper into this section, actually there are some problems:
1. If buffer is smaller than 8, it needs to append null, can't write just the characters.
2. Your comment said that if input is 8 long, it would add a null past the buffer. But with strncpy, that's wrong, it would only write the 8 characters in the buffer and stop there (resulting in an unterminated sequence, which fails to be a C string). Anyway if the compiler complains, you did right to change away from strncpy.

I would suggest perhaps to set the union to 0 on the numeric view, before calling memcpy. This guarantees no garbage nonzeros near the end of the buffer, and more predictable results. Or just use name.length() + 1, which ought to be safe, because c_str always points to a null terminated string.

Normally at this point I would take your branch, apply my fixes and close your PR with "done", but I'm not at the computer this weekend, so maybe if you can fix this, if all good, I'll be able to merge from my phone.